tests/lsp_features/completion.zig

```diff
@@ -1393,6 +1393,56 @@ test "builtin fn `field`" {
         .{ .label = "chip1fn1", .kind = .Function, .detail = "fn chip1fn1() void" },
         .{ .label = "chip1fn2", .kind = .Function, .detail = "fn chip1fn2(_: u32) void" },
     });
+    try testCompletion(
+        \\pub const chip_mod = struct {
+        \\    pub const devices = struct {
+        \\        pub const chip1 = struct {
+        \\            canary: bool,
+        \\            pub const peripherals = struct {};
+        \\            pub fn chip1fn1() void {}
+        \\            pub fn chip1fn2(_: u32) void {}
+        \\        };
+        \\        pub const chip2 = struct {
+        \\            pub fn chip2fn1() void {}
+        \\        };
+        \\    };
+        \\};
+        \\test {
+        \\    const chip_name = "chip1";
+        \\    const chip = @field(chip_mod.devices, chip_name);
+        \\    chip.<cursor>
+        \\}
+    , &.{
+        .{ .label = "peripherals", .kind = .Constant, .detail = "const peripherals = struct" },
+        .{ .label = "chip1fn1", .kind = .Function, .detail = "fn chip1fn1() void" },
+        .{ .label = "chip1fn2", .kind = .Function, .detail = "fn chip1fn2(_: u32) void" },
+    });
+    try testCompletion(
+        \\pub const chip_mod = struct {
+        \\    pub const devices = struct {
+        \\        pub const @"chip1" = struct {
+        \\            canary: bool,
+        \\            pub const peripherals = struct {};
+        \\            pub fn chip1fn1() void {}
+        \\            pub fn chip1fn2(_: u32) void {}
+        \\        };
+        \\        pub const chip2 = struct {
+        \\            pub fn chip2fn1() void {}
+        \\        };
+        \\    };
+        \\};
+        \\test {
+        \\    const chips = struct {
+        \\          pub const chip_name: []const u8 = "chip1";
+        \\      };
+        \\    const chip = @field(chip_mod.devices, chips.chip_name);
+        \\    chip.<cursor>
+        \\}
+    , &.{
+        .{ .label = "peripherals", .kind = .Constant, .detail = "const peripherals = struct" },
+        .{ .label = "chip1fn1", .kind = .Function, .detail = "fn chip1fn1() void" },
+        .{ .label = "chip1fn2", .kind = .Function, .detail = "fn chip1fn2(_: u32) void" },
+    });
```

Comment:
I think these tests are unnecesserily long. How about doing this:

Note that in the third test, the name of `Inner` is not `@"Inner"`. See the other review comment for the reasoning.
```zig
    try testCompletion(
        \\const Outer = struct {
        \\    const Inner = struct {
        \\        fn foo() void {}
        \\    };
        \\};
        \\test {
        \\    const chip = @field(Outer, "Inner");
        \\    chip.<cursor>
        \\}
    , &.{
        .{ .label = "foo", .kind = .Function, .detail = "fn foo() void" },
    });
    try testCompletion(
        \\const Outer = struct {
        \\    const Inner = struct {
        \\        fn foo() void {}
        \\    };
        \\};
        \\test {
        \\    const chip_name = "Inner";
        \\    const chip = @field(Outer, chip_name);
        \\    chip.<cursor>
        \\}
    , &.{
        .{ .label = "foo", .kind = .Function, .detail = "fn foo() void" },
    });
    try testCompletion(
        \\const Outer = struct {
        \\    const Inner = struct {
        \\        fn foo() void {}
        \\    };
        \\};
        \\test {
        \\    const chips = struct {
        \\        pub const chip_name: []const u8 = "Inner";
        \\    };
        \\    const chip = @field(Outer, chips.chip_name);
        \\    chip.<cursor>
        \\}
    , &.{
        .{ .label = "foo", .kind = .Function, .detail = "fn foo() void" },
    });
```

Reply:
There should be at least one `@"chip-name"` test to cover eg `pub const @"ESP32-C3" = struct {...};`.

Reply:
You mean something like this?
```zig
    try testCompletion(
        \\const Outer = struct {
        \\    const Inner = struct {
        \\        fn foo() void {}
        \\    };
        \\};
        \\test {
        \\    const chips = struct {
        \\        pub const @"chip-name": []const u8 = "Inner";
        \\    };
        \\    const chip = @field(Outer, chips.@"chip-name");
        \\    chip.<cursor>
        \\}
    , &.{
        .{ .label = "foo", .kind = .Function, .detail = "fn foo() void" },
    });
```
This doesn't seem to be related to `@field` but rather resolving `chips.@"chip-name"` itself.  #1574 has test coverage for this. But more tests will never hurt so do as you wish.

Reply:
True, but those tests didn't exist at the time, would probably keep them as they are.